pandas/core/categorical.py

```diff
@@ -1991,8 +1972,7 @@ def is_dtype_equal(self, other):
         """
 
         try:
-            return (self.categories.equals(other.categories) and
-                    self.ordered == other.ordered)
+            return hash(self.dtype) == hash(other.dtype)
```

Comment:
Can't this use the `__eq__` method of the dtype ?

Reply:
This would run into the strangeness of `CategoricalDtype(None) == CategoricalDtype(['a', 'b'], True)` being true, which we don't want here.

Reply:
Ah, that's a good point. But is there then a way for a user to do this, apart from this hash checking?

Reply:
>   But is there then a way for a user to do this, apart from this hash checking?

What does "this" mean? Checking if two `CategoricalDtypes` are the same?

Reply:
Yeah, in that case they would have to use the same "trick" of checking that the hashes are equal.

I expect comparing two instances `CategoricalDtype` to be relatively rare / for advanced users. In almost all of these cases, I expect users to be doing `a.dtype == b.dtype`, where `a` and `b` are `Categorical`, in which case they *won't* hit this gotcha, since a `Categorical().dtype` will never have unknown categories. It's just "free-standing" `CategoricalDtype`, not attached to a `Categorical`, that can have unknown categories.

Reply:
Indeed. And as an advanced user you can always make sure to avoid having dtype objects without categories